### PR TITLE
Stripped off the release number from pkg name before checking

### PR DESCRIPTION
**Pull request by tinawang123**
Before checking, the release number needs to be stripped off of
the package name according bsc#1179303
Refer jira feature:  https://jira.suse.com/browse/SLE-8608

- Related ticket: https://progress.opensuse.org/issues/80588
- Verification run: https://openqa.suse.de/tests/5188020

### CODE REVIEW

File: lib/services/hpcpackage_remain.pm [hunk 1]
```diff
@@ -53,17 +51,17 @@ sub compare_pkg {
...
 sub check_pkg {
-    my @pkglist = split('\n', script_output("rpm -qa", proceed_on_failure => 1, timeout => 180));
-    my %hash_a  = map { $_ => 1 } @pkglist;
-    my %hash_b  = map { $_ => 1 } @diffpkg;
-    diag 'After: ' . Dumper(\@pkglist);
-    diag @diffpkg;
-    my @b_only = grep { !$hash_a{$_} } @diffpkg;
+    my $arch    = get_var('ARCH');
+    my @pkglist = split('\n', script_output("rpm -qa | tee /tmp/afterpkg.txt", proceed_on_failure => 1, timeout => 180));
+    my $before  = '/tmp/diffpkg.txt';
```
Comment (lemon-suse) [inline lib/services/hpcpackage_remain.pm:61]: It is better to set global variable for before and after, maybe you can name as before_pkgs and after_pkgs.

**Comment by lemon-suse** [comment]
For the PR title, how do you think change it to 'Stripped off the release number from pkg name before checking', and could you set it with the feature number in the title and description which may make reviewer clear what you are fixing, change request also include the commit title and description.

File: lib/services/hpcpackage_remain.pm [hunk 2]
```diff
@@ -53,17 +51,17 @@ sub compare_pkg {
...
+    my $after   = '/tmp/afterpkg.txt';
+    my $out     = script_output("diff $before $after | grep -iE '*.-[[:digit:].]$arch'", proceed_on_failure => 1, timeout => 180);
```
Comment (coolgw) [inline lib/services/hpcpackage_remain.pm:63]: this only make filter on those diff string with digital suffix, but not Fuzzy match

File: lib/services/hpcpackage_remain.pm [hunk 3]
```diff
@@ -53,17 +51,35 @@ sub compare_pkg {
...
-    my %hash_b  = map { $_ => 1 } @diffpkg;
-    diag 'After: ' . Dumper(\@pkglist);
-    diag @diffpkg;
-    my @b_only = grep { !$hash_a{$_} } @diffpkg;
+    diag "Before migration: " . Dumper(\@diffpkg);
+    diag "After migration:" . Dumper(\@pkglist);
+    my @after  = del_num(@pkglist);
+    my @before = del_num(@diffpkg);
+    my %hash_a = map  { $_ => 1 } @after;
+    my %hash_b = map  { $_ => 1 } @before;
```
Comment (coolgw) [inline lib/services/hpcpackage_remain.pm:81]: we do not need hash_b

File: lib/services/hpcpackage_remain.pm [hunk 4]
```diff
@@ -53,17 +51,35 @@ sub compare_pkg {
     $list1 //= '';
     $list2 //= '';
     # Make a list of installed packages and diff against list from installed packages
-    @diffpkg = split('\n', script_output("diff $list1 $list2 | grep -E '^>' | cut -d' ' -f2 |  grep -E '.*-hpc.*' |   grep -vE 'system|module|suse' |  grep -E '.*_[[:digit:]]+_[[:digit:]]+.*-gnu|.*_[[:digit:]]+_[[:digit:]]+.*-hpc' |  grep -vE '.*-static\$' | tee /tmp/diffpkg.txt", proceed_on_failure => 1, timeout => 180));
+    @diffpkg = split('\n', script_output("diff $list1 $list2 | grep -E '^>' | cut -d' ' -f2 |  grep -E '.*-hpc.*' |   grep -vE 'system|module|suse' |  grep -E '.*_[[:digit:]]+_[[:digit:]]+.*-gnu|.*_[[:digit:]]+_[[:digit:]]+.*-hpc' |  grep -vE '.*-static\$'", proceed_on_failure => 1, timeout => 180));
+}
+
+# The release number needs to be stripped off of the package name
+sub del_num {
```
Comment (coolgw) [inline lib/services/hpcpackage_remain.pm:58]: this function stripped off not only the sub release number , but also remove the arch part.

Comment (tinawang123) [reply lib/services/hpcpackage_remain.pm:58]: Yes, It's OK. As they all have same arch. We don't need compare this part.

File: lib/services/hpcpackage_remain.pm (hunk 3 above)
Comment (tinawang123) [reply lib/services/hpcpackage_remain.pm:81]: I will remove this one.

**Review by lemon-suse — APPROVED** [review]

**Pull request merged**